datafusion/expr/src/expr_schema.rs

```diff
@@ -352,7 +358,18 @@ impl ExprSchemable for Expr {
                     }
                 }
                 WindowFunctionDefinition::AggregateUDF(func) => Ok(func.is_nullable()),
-                WindowFunctionDefinition::WindowUDF(udwf) => Ok(udwf.nullable()),
+                WindowFunctionDefinition::WindowUDF(udwf) => {
+                    let data_types = args
+                        .iter()
+                        .map(|e| e.get_type(input_schema))
+                        .collect::<Result<Vec<_>>>()?;
+                    let input_types = data_types_with_window_udf(&data_types, udwf)?;
```

Comment:
Note: I copied the call to type-coercion performed for window function arguments from `Expr::get_type` (above).

Reply:
Sounds good. I'll update the PR with the suggested change.